### PR TITLE
ci: add semantic pull request linting

### PR DESCRIPTION
**Pull request by jooola**
This enforce semantic pull request messages before merging.

Do we want to require a scope or do we allow un-scoped prefixes ?
Did I miss a scope in the allowed scopes ?

The subjects should not start with a uppercase letter.

### CODE REVIEW

File: .github/workflows/lint-pr.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            analyzer
+            api
+            api_client
+            installer
+            legacy
+            playout
+            ui
+            worker
+
+            ci
```
Comment (paddatrapper) [inline .github/workflows/lint-pr.yml:29]: ci should be a type, not a scope

File: .github/workflows/lint-pr.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          scopes: |
+            analyzer
+            api
+            api_client
+            installer
+            legacy
+            playout
+            ui
+            worker
+
```
Comment (paddatrapper) [inline .github/workflows/lint-pr.yml:28]: Is this intentional to allow blank scope?

File: .github/workflows/lint-pr.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            ci
+          subjectPattern: ^(?![A-Z]).+$
+          subjectPatternError: |
+            The subject "{subject}" found in the pull request title "{title}"
+            didn't match the configured pattern. Please ensure that the subject
+            doesn't start with an uppercase character.
```
Comment (paddatrapper) [inline .github/workflows/lint-pr.yml:34]: Maybe describe or link to expected format?

File: .github/workflows/lint-pr.yml (hunk 2 above)
Comment (jooola) [reply .github/workflows/lint-pr.yml]: Nope, I wanted to split our project components  from abstract scope, but I just found out ci is actually a type.

File: .github/workflows/lint-pr.yml (hunk 1 above)
Comment (jooola) [reply .github/workflows/lint-pr.yml]: Yup, noted

**Comment by jooola** [comment]
Fixed the few comments

**Comment by jooola** [comment]
I feel like we should require a scope for every Pr commit messages. As it is difficult not to forget a scope for some component specific changes. Looking at the history we merged some commits with the scope 'shared' some without but should have.

If we don't enforce such structure we will either have to sort the changes logs in each scope/component, or we don't care about having changes structured in a coped sections.

**Comment by paddatrapper** [comment]
I have no strong feelings either way, so happy to enforce it

**Pull request merged**